### ISSUE TITLE
Item insertion via layout updating broken?

### ISSUE THREAD

**Issue by lgrapenthin** [STRML/react-grid-layout#262]
I have recently switched from a 0.9.0 to the current version 0.12.16.

When we insert new items RRGL (now with Widthprovider) doesn't position them according to the layout we provide simultaneously with the new children, but instead always appends new items to the bottom left corner in a new row even though our layout specifies a different position. It also ignores the fact that our layout has moved an item to make space for the newly added item. The item that we moved stays at the same position and the new item is added appended at the bottom left corner.

If we re-mount the component e. g. via page reload it works as before and the new item is where it belongs. We have checked that the layout we generate is equal in both cases (update / re-mount) to make sure that we didn't rely on broken behavior of the old version. Keys are all set correctly etc.

I found many issues here regarding layout updating that I fail to understand, so before creating a minimal usecase I wanted to ask whether this is a known problem?

TL;DR - If the layout prop is changed to layout x the grid looks different than it would be it had been initialized with layout x. This must have changed since 0.9.0

**Comment by lgrapenthin** [comment]
@STRML Is there a hack or workaround you could recommend or should we just downgrade?

**Comment by lgrapenthin** [comment]
issue doesn't happen with 0.12.7
